astropy/stats/sigma_clipping.py

```diff
@@ -194,6 +209,23 @@ def perform_clip(_filtered_data, _kwargs):
         warnings.warn("Input data contains invalid values (NaNs or infs), "
                       "which were automatically masked.", AstropyUserWarning)
 
+    if HAS_SCIPY is True:
+        if masked is False:
+            if cenfunc is not np.ma.mean or stdfunc is not np.std:
```

Comment:
I think I'd agree about it raising an exception here as if someone changes cenfunc or stdfunc from the default,  they would be expecting those to be used